front/pages/poke/[wId]/upgrade_enterprise/index.tsx

```diff
@@ -0,0 +1,303 @@
+
+  return {
+    props: {
+      owner,
+    },
+  };
+});
+
+export default function NewEnterpriseSubscriptionForm({
+  owner,
+}: {
+  owner: WorkspaceType;
+}) {
+  const router = useRouter();
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [error, setError] = React.useState<string | null>(null);
+
+  const form = useForm<NewEnterpriseSubscriptionType>({
+    resolver: ioTsResolver(NewEnterpriseSubscriptionSchema),
+    defaultValues: {
+      stripeSubscriptionId: "",
+      code: "",
+      name: "",
+      isSlackbotAllowed: true,
+      isSlackAllowed: true,
+      isNotionAllowed: true,
+      isGoogleDriveAllowed: true,
+      isGithubAllowed: true,
+      isIntercomAllowed: true,
+      isConfluenceAllowed: true,
+      isWebCrawlerAllowed: true,
+      maxMessages: -1,
+      dataSourcesCount: -1,
+      dataSourcesDocumentsCount: -1,
+      dataSourcesDocumentsSizeMb: 2,
+      maxUsers: 1000,
+    },
+  });
+
+  const onSubmit = (values: NewEnterpriseSubscriptionType) => {
+    const cleanedValues = Object.fromEntries(
+      removeNulls(
+        Object.entries(values).map(([key, value]) => {
+          if (typeof value !== "string") {
+            return [key, value];
+          }
+          const cleanedValue = value.trim();
+          if (!cleanedValue) {
+            return null;
+          }
+          return [key, cleanedValue];
+        })
+      )
+    );
+
+    void submit();
+    async function submit() {
+      setIsSubmitting(true);
+      try {
+        const r = await fetch(
+          `/api/poke/workspaces/${owner.sId}/upgrade_enterprise`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify(cleanedValues),
+          }
+        );
+
+        if (!r.ok) {
+          throw new Error(
+            `Something went wrong: ${r.status} ${await r.text()}`
+          );
+        }
+
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        void router.push(`/poke/${owner.sId}`);
+      } catch (e) {
+        setIsSubmitting(false);
+        if (e instanceof Error) {
+          setError(e.message);
+        }
+      }
+    }
+  };
+
+  console.log("Form errors", form.formState.errors);
+
+  return (
+    <div className="min-h-screen bg-structure-50 pb-48">
+      <PokeNavbar />
+      <div className="mx-auto h-full max-w-2xl flex-grow flex-col items-center justify-center pt-8">
+        <PokeForm {...form}>
+          <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-8">
+            <InputField
+              control={form.control}
+              name="stripeSubscriptionId"
+              title="Stripe Subscription id"
+              placeholder="sub_1234567890"
+            />
+            <InputField
+              control={form.control}
+              name="code"
+              title="Plan Code"
+              placeholder="ENT_COMPANY_NAME_PLAN_NAME"
```

Comment:
We kind probably use more user-friendly placeholders, no?

Reply:
I can put ENT_Soupinou_MAU10_FLOOR500 so that it looks like our current convention!